### PR TITLE
Shenli/set password

### PR DESCRIPTION
**Pull request by shenli**

### CODE REVIEW

File: parser/parser_test.go [hunk 1]
```diff
@@ -203,6 +203,9 @@ func (s *testParserSuite) TestParser0(c *C) {
 		// SET CHARACTER SET
 		{"SET CHARACTER SET utf8mb4;", true},
 		{"SET CHARACTER SET 'utf8mb4';", true},
+		// Set password
+		{"SET PASSWORD = 'password';", true},
+		{"SET PASSWORD FOR 'shenli'@'localhost' = 'password';", true},
```
Comment (siddontang) [inline parser/parser_test.go]: I think root may be better than your name......

File: mysqldef/const.go [hunk 2]
```diff
@@ -115,3 +115,13 @@ const (
 const (
 	AuthName = "mysql_native_password"
 )
+
+// mysql database and tables
```
Comment (qiuyesuifeng) [inline mysqldef/const.go]: MySQL

File: stmt/stmts/set.go [hunk 3]
```diff
@@ -230,6 +235,30 @@ func (s *SetPwdStmt) SetText(text string) {
...
+				Schema: model.NewCIStr(mysql.MySQLDB),
+			},
+		},
+	}
+	asgn := expressions.Assignment{
+		ColName: "Password",
+		Expr:    expressions.Value{Val: s.Password},
+	}
+	nameMatch := expressions.NewBinaryOperation(opcode.EQ, &expressions.Ident{CIStr: model.NewCIStr("User")}, &expressions.Value{Val: userName})
+	hostMatch := expressions.NewBinaryOperation(opcode.EQ, &expressions.Ident{CIStr: model.NewCIStr("Host")}, &expressions.Value{Val: host})
```
Comment (qiuyesuifeng) [inline stmt/stmts/set.go]: Here not check name or host match, how about nameExpr and hostExpr?

File: stmt/stmts/set.go [hunk 4]
```diff
@@ -230,6 +235,30 @@ func (s *SetPwdStmt) SetText(text string) {
...
+	where := expressions.NewBinaryOperation(opcode.AndAnd, nameMatch, hostMatch)
+	st := &UpdateStmt{
+		TableRefs: r,
+		List:      []expressions.Assignment{asgn},
+		Where:     where,
+	}
+	return st.Exec(ctx)
```
Comment (qiuyesuifeng) [inline stmt/stmts/set.go]: Btw, how do u think about building an update sql to excute?

Comment (shenli) [inline stmt/stmts/set.go]: I don't want to call parser for the second time.

File: stmt/stmts/set.go (hunk 3 above)
Comment (shenli) [inline stmt/stmts/set.go]: You mean check the validation of name/host?
The validation will be checked in update.Exec(). The privilege check will be coming later.

**Comment by shenli** [comment]
PTAL

File: stmt/stmts/set.go (hunk 3 above)
Comment (qiuyesuifeng) [inline stmt/stmts/set.go]: I only mean whether the variable names use nameExpr and hostExpr are better, for nameMatch and nameMatch sound like boolean variables.

Comment (qiuyesuifeng) [inline stmt/stmts/set.go]: I only mean whether the variable names use nameExpr and hostExpr are better, for nameMatch and nameMatch sound like boolean variables.

File: stmt/stmts/set.go (hunk 4 above)
Comment (qiuyesuifeng) [inline stmt/stmts/set.go]: Gotcha.

File: stmt/stmts/set.go (hunk 3 above)
Comment (shenli) [inline stmt/stmts/set.go]: They are boolean expressions. For nameMatch means if "User" equals userName.

**Comment by qiuyesuifeng** [comment]
LGTM

File: stmt/stmts/create_test.go [hunk 5]
```diff
@@ -32,18 +33,20 @@ var _ = Suite(&testStmtSuite{})
...
-	dropDBSql      string
-	useDBSql       string
-	createTableSql string
-	insertSql      string
-	selectSql      string
+	testDB                  *sql.DB
+	createDBSql             string
+	dropDBSql               string
+	useDBSql                string
+	createTableSql          string
```
Comment (ngaut) [inline stmt/stmts/create_test.go]: Replace Sql to Text would be better.

File: stmt/stmts/create_test.go [hunk 6]
```diff
@@ -59,6 +62,12 @@ func (s *testStmtSuite) SetUpTest(c *C) {
 	s.selectSql = `SELECT * from test limit 2;`
 	mustExec(c, s.testDB, s.createDBSql)
 	mustExec(c, s.testDB, s.useDBSql)
+
+	s.createMySQLDBSQL = fmt.Sprintf("create database if not exists %s;", mysql.MySQLDB)
```
Comment (ngaut) [inline stmt/stmts/create_test.go]: MySQLDBSQL looks wired. Could you rename createMySQLDBSQL to createDBText

File: stmt/stmts/create_test.go (hunk 5 above)
Comment (shenli) [inline stmt/stmts/create_test.go]: They are all SQL statements.

File: stmt/stmts/create_test.go (hunk 6 above)
Comment (shenli) [inline stmt/stmts/create_test.go]: MySQL is the DB name. This is the system db. How about createSystemDB?

File: stmt/stmts/create_test.go (hunk 5 above)
Comment (ngaut) [inline stmt/stmts/create_test.go]: Yes, SQL is structured query language, it doesn't means text.
But we do mean text here.

File: stmt/stmts/create_test.go (hunk 6 above)
Comment (ngaut) [inline stmt/stmts/create_test.go]: createSystemDB is ok for me :)

**Comment by shenli** [comment]
PTAL

**Comment by ngaut** [comment]
LGTM

**Pull request merged**